core/storage/mpool/mpool.cpp

```diff
@@ -541,6 +541,7 @@ namespace fc::storage::mpool {
           [&](auto, auto bls, auto &cid, auto *smsg, auto *msg)
               -> outcome::result<void> {
             if (bls) {
+              std::shared_lock bls_cache_lock{bls_cache_mutex_};
```

Comment:
cache read mutates, to reorder elements
```suggestion
              std::unique_lock bls_cache_lock{bls_cache_mutex_};
```